newIDE/app/src/MainFrame/index.js

```diff
@@ -2398,6 +2427,61 @@ const MainFrame = (props: Props) => {
           // We don't block the save, as the user may want to save it anyway.
         }
 
+        if (storageProviderInternalName === 'Cloud') {
```

Comment:
Two things makes me cry at night:
- `if` that checks a specific provider (notably `=== 'Cloud'`)
- any new line of code, notably if it contains logic, being added to MainFrame ;) 

==> Let's move this to a "storageProviderOperations" or something else so we can do `await storageProviderOperations.checkProjectCanBeSafelySaved(project, currentFileMetadata)` 

(unsure about the name and parameters, just to give you the idea)

Reply:
This has been moved 👍